### PR TITLE
Initial /aggregate implementation, for init requests.

### PR DESCRIPTION
**Pull request by BranLwyd**
Apologies for the size of this PR; a review strategy that might work is:
1) Review the datastore/schema changes first.
2) Once you are happy with the datastore changes, review the aggregator changes.

I need to land https://github.com/divviup/libprio-rs/pull/202 before we can commit this, but it seems very likely that any changes required from review of that PR will be minimal. I want to send this for review ASAP since it has already taken quite a while & grown quite large.

### CODE REVIEW

File: janus_server/src/aggregator.rs
Comment (tgeoghegan) [inline janus_server/src/aggregator.rs:377]:
```suggestion
                    "VDAF took an inconsistent number of rounds to reach Finish state"
```

**Review by tgeoghegan — CHANGES_REQUESTED** [review]

**Review by tgeoghegan — APPROVED** [review]
It's going to take me a little time to make the requested changes on #37, so I'm marking this approved so that you can go ahead and merge this when you think it's ready and then I'll rebase #37 as needed.

File: janus_server/src/bin/aggregator.rs
Comment (BranLwyd) [reply janus_server/src/bin/aggregator.rs:44]: It's going to take some time before this is ready to merge (the dependent `libprio-rs` PR has at least one round of communication with cjpatton to go before it can possibly be merged), so let's rebase based on whoever gets to `main` first :-)

File: janus_server/src/datastore.rs
Comment (BranLwyd) [reply janus_server/src/datastore.rs:233]: I agree. This is a general datastore change (several existing funcs also need to be touched), and I've got a few other "small" datastore cleanups in mind; unless you object, I'm going to make this change in a followup PR.

File: janus_server/src/aggregator.rs
Comment (BranLwyd) [reply janus_server/src/aggregator.rs:265]: I disagree, this data-type is specific to this method & I don't think there's much reason to widen the visibility -- the upside is basically nil, the downside is that the code is slightly harder to understand because readers have to realize that `ReportShareData` is only used here. [If more than one method wanted to use this type, then yeah, I'd support moving it out -- but this is so situationally-specific that I don't know we'll need exactly the same type.]

I minimalized the trait bounds, but unfortunately I still need one bound.

File: janus_server/src/aggregator.rs
Comment (BranLwyd) [reply janus_server/src/aggregator.rs]: You're correct, of course. I created a `test_util::fake::Vdaf` class that is a minimal implementation of the `vdaf::{Vdaf,Aggregator,Client}` traits with all of the implementations required for a VDAF to be usable by `janus`, and implemented a `prepare_init` failure test (and added a `prepare_step` failure test too, since it was only a small amount of additional work).

Other than being useful for testing, this fake might be the best indicator of what implementations are practically required for a VDAF to work with `janus`. As it is, we're already at ~100 lines of code even for a do-nothing implementation!

**Pull request merged**
